### ISSUE TITLE
How many positive and negative ref-bboxes do you use in each batch during training?

### ISSUE THREAD

**Issue by hellowangqian** [superxuang/caffe_3d_faster_rcnn#11]
Nice work indeed. I have two questions regarding the implementation details.

You mentioned in the last paragraph of Section III B in your paper :  “4 base sizes are used” and “in each spatial dimension to define 64 reference B-boxes at each feature map cell”.
May you clarify how these 64 reference B-boxes are defined based on 4 base sizes?

My second question is related to Eq(4-5) in your paper. You  must have much more negative reference B-boxes than positive ones, so how do you select them for the loss computation during training.
looking forward to your answer.
Thanks.

**Comment by superxuang** [comment]
Hello, thank you very much for your questions.

For your first question, we generate 64 reference B-boxes by combining 4 base sizes in 3 dimensions (i.e. x (width), y (height), and z (length)). For example, when we use 4 base sizes [30, 60, 120, 240], we can get 4^3 = 64 different reference B-boxes with the shape of [30 x 30 x 30], ..., [30 x 30 x 240], ... , [30 x 240 x 240], ... , [240 x 240 x 240].

For your second question about the inter-class imbalance, in the classification loss function (i.e. Eq 4), we calculate the instance number _N_ of each class and inverse the instance number _1/N_ as the weight of this class to rebalance different classes. In the regression loss function (i.e. Eq 5), we also use an inversed instance number _1/N_ to rebalance different foreground classes (background class is excluded in the regression loss function).

Thanks again for your question.

**Comment by hellowangqian** [comment]
Thank you very much for your clear answers!
